src/snap_management.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Learn more at: https://juju.is/docs/sdk
+
+"""Snap Installation Module.
+
+Modified from https://github.com/canonical/k8s-operator/blob/main/charms/worker/k8s/src/snap.py
+"""
+
+
+import logging
+import platform
+
+import charms.operator_libs_linux.v2.snap as snap_lib
+from charms.operator_libs_linux.v2.snap import SnapState
+
+# Log messages can be retrieved using juju debug-log
+log = logging.getLogger(__name__)
+
+
+# Map of the grafana-agent snap revision to install for given architectures and strict mode.
+_grafana_agent_snap_spec = {
```

Comment:
I think I see where this is coming from, but imho I think it'd be cleaner to do:
```python
_strict_amd64_gagent_revision = 16
_strict_arm64_gagent_revision = 42

def install_ga_snap(classic):
    if not classic:
        if arch == "amd64":
            rev = _strict_amd64_gagent_revision
        elif  arch == "arm64":
            rev = _strict_arm64_gagent_revision
        else:
            raise UnsupportedArch(arch)
    else:
        ???
```

feels more explicit this way, rather than having a massive untyped data structure whose only purpose is to map a few string params into a single integer

Reply:
Not against this change, but I think it'll get messy once we have the classic snap revs too.  Something like:

```python
_strict_amd64_gagent_revision = 16
_strict_arm64_gagent_revision = 42
_classic_amd64_gagent_revision = 987
_classic_arm64_gagent_revision = 123

def install_ga_snap(classic):
    if not classic:
        if arch == "amd64":
            rev = _strict_amd64_gagent_revision
        elif  arch == "arm64":
            rev = _strict_arm64_gagent_revision
        else:
            raise UnsupportedArch(arch)
    else:
        if arch == "amd64":
            rev = _classic_amd64_gagent_revision
        elif  arch == "arm64":
            rev = _classic_arm64_gagent_revision
        else:
            raise UnsupportedArch(arch)
```

Which I think is less readable than the current state.  I'm flexible though if others disagree

Reply:
Starts to look like embedded C code :S

Reply:
Updated this based on an offline conversation with @PietroPasotti to store the manifest as:
```python
_grafana_agent_snaps = {
    # (confinement, arch): revision
    ("strict", "amd64"): 16,
    ("strict", "arm64"): 23,
}
```